### PR TITLE
Update codeQL to v3

### PR DESCRIPTION
**Pull request by spolti**
chore:	Update CodeQL to V3 to get rid of this warning:
	`Warning: CodeQL Action v2 will be deprecated on December 5th, 2024`
	Plus, attenpt to fix the Snyk Container scan failures due errors when trying to
	upload the SERIF file:
```
Processing sarif files: ["application/storage-initializer/docker.snyk.sarif"]
  Uploading results
  Successfully uploaded results
Waiting for processing to finish
Error: Code Scanning could not process the submitted SARIF file:
could not convert rules: invalid security severity value, is not a number: null
ConfigurationError: Code Scanning could not process the submitted SARIF file:
could not convert rules: invalid security severity value, is not a number: null
    at run (/home/runner/work/_actions/github/codeql-action/v2/lib/upload-sarif-action.js:65:15)
```

**Release note**:

```release-note
None
```

### CODE REVIEW

File: .github/workflows/scheduled-image-scan.yml [hunk 1]
```diff
@@ -45,7 +45,8 @@ jobs:
 
       - name: Upload sarif file to Github Code Scanning
         if: always()
-        uses: github/codeql-action/upload-sarif@v2
+        continue-on-error: true #avoid fail the pipeline if the SARIF upload fails.
```
Comment (terrytangyuan) [inline .github/workflows/scheduled-image-scan.yml:48]: What are some example upload failures?

Comment (spolti) [reply .github/workflows/scheduled-image-scan.yml:48]: Hi @terrytangyuan, here it is: https://github.com/kserve/kserve/actions/runs/8405425651

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Pull request merged**